### PR TITLE
Added DID Configuration Verification Endpoint

### PR DESCRIPTION
**Pull request by andresuribe87**
# Overview
This PR adds the `v1/did-configurations/verification` endpoint, with documentation. It follows the spec in https://identity.foundation/.well-known/resources/did-configuration/#did-configuration-resource-verification

# Description
Now that we have an endpoint that can create a DIDConfiguration, this PR adds another endpoint so that clients can verify they've done the correct steps when hosting it (or that any other DID Configuration is correct).

# How Has This Been Tested?
Unit tests were added. Integration tests were not added, since verifying `https://www.tbd.website` fails with the following reason : `verifying JWT credential: error getting key to verify credential<>: did<did:key:z6Mkh4A6QTNCAFi4NZfnWt8qNSGXDGnNXhxXWewcpBrzS19v> has no verification methods with kid: did:key:z6Mkh4A6QTNCAFi4NZfnWt8qNSGXDGnNXhxXWewcpBrzS19v`. I.e. we need to update the JWT to use the verification method ID, instead of the DID. This is tracked in a separate issue. 

# Checklist

## References
https://identity.foundation/.well-known/resources/did-configuration/#did-configuration-resource-verification

### CODE REVIEW

File: pkg/service/well-known/did_configuration.go [hunk 1]
```diff
@@ -33,6 +56,80 @@ func (s DIDConfigurationService) Status() svcframework.Status {
...
+
+	response := VerifyDIDConfigurationResponse{
+		DIDConfiguration: bodyCopy.String(),
+	}
+	for _, domainLinkageCredential := range didConfiguration.LinkedDIDs {
+		// For a Domain Linkage Credential to be deemed valid, it MUST be successfully processed in accordance with the following steps:
+		//
+		// 1. The credentialSubject.id MUST be a DID, and the value MUST be equal to both the Subject and Issuer of the Domain Linkage Credential.
+		credentialSubjectID := domainLinkageCredential.Credential.CredentialSubject["id"].(string)
+		if !strings.HasPrefix(credentialSubjectID, "did:") {
```
Comment (decentralgabe) [inline pkg/service/well-known/did_configuration.go:97]: believe there's a method `IsValidDID` or similar in the utils somewhere

File: pkg/service/well-known/did_configuration.go [hunk 2]
```diff
@@ -1,24 +1,47 @@
...
 	keyStoreService *keystore.Service
+	validator       *credint.Validator
+
+	HTTPClient *http.Client
 }
 
-func NewDIDConfigurationService(keyStoreService *keystore.Service) *DIDConfigurationService {
-	return &DIDConfigurationService{keyStoreService: keyStoreService}
+func NewDIDConfigurationService(keyStoreService *keystore.Service, didResolver resolution.Resolver, schema *schema.Service) (*DIDConfigurationService, error) {
+	client := &http.Client{Transport: otelhttp.NewTransport(http.DefaultTransport)}
```
Comment (decentralgabe) [inline pkg/service/well-known/did_configuration.go:34]: should this be the default client (with timeouts, etc) and then add in the otel?

File: pkg/service/well-known/did_configuration.go (hunk 1 above)
Comment (andresuribe87) [reply pkg/service/well-known/did_configuration.go:97]: I searched for something similar, but couldn't find anything.

File: pkg/service/well-known/did_configuration.go (hunk 2 above)
Comment (andresuribe87) [reply pkg/service/well-known/did_configuration.go:34]: DefaultClient is the same as the zero value of `http.Client`. Timeouts are carried by the `ctx` when the call is made. 

We _could_ have an `httpClientProvider`, but then that get's into the dependency injection and I wanted to keep this simple.

**Pull request merged**